util/grpc/src/retry_config.rs

```diff
@@ -0,0 +1,38 @@
+use retry::delay;
+use serde::Serialize;
+use std::time::Duration;
+use structopt::StructOpt;
+
+/// An object which represents a retry policy for retriable errors for a grpc
+/// connection
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Serialize, StructOpt)]
+pub struct GrpcRetryConfig {
+    /// How many times to retry when we get retriable errors (grpc connection)
+    #[structopt(long, env, default_value = "3")]
+    pub grpc_retry_count: usize,
+
+    /// How long to back off (milliseconds) when we get retriable errors (grpc
+    /// connection)
+    #[structopt(long, env, default_value = "20")]
+    pub grpc_retry_millis: u64,
+}
+
+impl Default for GrpcRetryConfig {
+    fn default() -> Self {
+        Self {
+            grpc_retry_count: 3,
+            grpc_retry_millis: 20,
+        }
+    }
+}
+
+impl GrpcRetryConfig {
+    /// Get a duration iterator for use with retry crate based on this config
+    pub fn get_retry_iterator(&self) -> Box<dyn Iterator<Item = Duration>> {
```

Comment:
IIUC this method can return `impl Iterator<Item = Duration>` instead of a `Box`

Reply:
I don't think so, I think that works on arguments to functions but not return values from functions.

We could make struct that implements `Iterator<Duration>` and return that to avoid the Box, that's what they do in the standard library for functions that return iterators. I just felt like the optimization is unnecessary.

Reply:
I'm pretty sure impl in return types is stabilized: https://rust-lang.github.io/impl-trait-initiative/explainer/rpit.html

I recommend you try it. Either way, I'm not attached to doing this, as the Box overhead is likely negligible compared to the actual RPCs/calls.